### PR TITLE
[DEV-7197] Spending by Award DEFC filter updates

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Adjusts the values returned by the `spending_by_award` endpoint to filter the Covid-19 related outlays and obligations by the provided DEFC filter.

**Technical details:**
If the COVID-19 Obligations or COVID-19 Outlays values are requested in a call to `spending_by_award`, the values returned will be filtered by the `def_codes` filter (if present). So if Award 1 has $100 in obligations under DEFC "L" and 
 $20 in obligations under DEFC "P", if the def_codes filter contains only "L", the value returned for Award 1's obligations will be $100, not $20.

**Requirements for PR merge:**

1. [ ] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [X] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-7197](https://federal-spending-transparency.atlassian.net/browse/DEV-7197):
    - [X] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/search/v2/views/spending_by_award.py [hunk 1]
```diff
@@ -343,6 +343,29 @@ def construct_es_response_for_prime_awards(self, response) -> dict:
...
+                        x["outlay"]
+                        if (self.filters.get("def_codes") is not None and x["defc"] in self.filters["def_codes"])
+                        or self.filters.get("def_codes") is None
+                        else 0
+                        for x in row.get("COVID-19 Outlays")
+                    ]
+                )
+            if row.get("def_codes"):
+                if self.filters.get("def_codes"):
+                    row["def_codes"] = filter(lambda x: x in self.filters.get("def_codes"), row["def_codes"])
```
Comment (sethstoudenmier) [inline usaspending_api/search/v2/views/spending_by_award.py:368]: You might need to wrap the `filter()` in `list()` because testing locally it appears that DRF is representing a filter object as an empty list.

Comment (sethstoudenmier) [reply usaspending_api/search/v2/views/spending_by_award.py:368]: Also, do we want to limit the DEFC returned to only those that are filtered on? Realized while working on the download counterpart that it wasn't mentioned in any AC and keep going back and forth on it. Originally was for limiting the DEFC shown to only those that are filtered to, but could come across as hiding information. My thinking is that Outlay and Obligation are different since on that page we don't list everything out, only a single number (unlike the list of DEFC next to it).

**Review by sethstoudenmier — CHANGES_REQUESTED** [review]
Obligations and Outlays appear to work as expected. The changes to `def_codes` in the response are always returning an empty list. I do wonder if we even want the changes to the `def_codes` returned though.

File: usaspending_api/search/v2/views/spending_by_award.py (hunk 1 above)
Comment (emily-brents) [reply usaspending_api/search/v2/views/spending_by_award.py:368]: Initially I was not going. to filter the DEFC array, but based on [this](https://data-transparency-bfs.slack.com/archives/C0122HUCD17/p1620831604086100) conversation with Beth, it seems that the PO wants the array to be filtered as well. I also am not seeing the bug you're talking about with the `def_codes` array, but I'll try and look into it some more.

Comment (sethstoudenmier) [reply usaspending_api/search/v2/views/spending_by_award.py:368]: >Initially I was not going. to filter the DEFC array, but based on this conversation with Beth, it seems that the PO wants the array to be filtered as well

Gotcha. I will make the change to the download as well to reflect this change.

**Review by sethstoudenmier — APPROVED** [review]
